Red thread (#89)
chhoumann/p7: report/sections/Chapter 01 - Introduction/Intro.tex

Therefore, many companies such as Meta and Galois use Haskell for cryptography, avionics monitoring and spam filtering. \todo{Insert citation}
Thus, Haskell and other languages within the functional programming paradigm are useful tools for software engineers.

Haskell is taught in the Programming Paradigms course on the 1st semester of the Software Master's Degree at \aau{}.
Most students attending this course failed in the past.
This resulted in a survey being sent out, and a student task force being assembled to understand the issues that caused the high failure rate.
Here, many students mentioned not feeling experienced enough with the programming languages used in the course to teach the functional and logical programming paradigms. \todo{Insert citation}

Consequently, \aau{} chose to restructure the course entirely.
Rather than teaching many different programming languages within the functional and logical programming paradigms, the course was restructured to focus only on Haskell.
Furthermore, the course now features more practical programming exercises with classroom discussion in-between, rather than traditional lectures followed by a separate exercise session.

> Supporting this way of teaching, the course lectures have been moved to a more classical classroom setup. \todo{Could be worded differently}

Reviewer: ```suggestion
To accommodate the course changes, the course lecture have been changed into a classroom setup where lecturing and exercises are interweaved.
```

The feedback loop for each platform is quite different, however.

> Codewars focuses on competitive programming, meaning users compete to solve problems fastest and most efficient way. \todo{Spelling}

Reviewer: ```suggestion
Codewars focuses on competitive programming where users compete to solve problems.
```
